examples/fft_vs_geom_movie.py

```diff
@@ -0,0 +1,348 @@
+# Copyright (c) 2012-2016 by the GalSim developers team on GitHub
+# https://github.com/GalSim-developers
+#
+# This file is part of GalSim: The modular galaxy image simulation toolkit.
+# https://github.com/GalSim-developers/GalSim
+#
+# GalSim is free software: redistribution and use in source and binary forms,
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions, and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions, and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+#
+
+"""@file fft_vs_geom_movie.py
+Script to demonstrate the geometric approximation to Fourier optics.  Simulates both a randomly
+evolving set of Zernike aberrations and a frozen-flow Kolmogorov atmospheric phase screen.
+
+To visualize only the Zernike aberrations, add `--nlayers 0` as a command line argument.
+
+To visualize only atmospheric effects, add `--sigma 0` as a command line argument.
+
+Some suggested command lines, appropriate for LSST:
+
+Optics only simulation:
+  $ python fft_vs_geom_movie.py --diam 8.36 --obscuration 0.61 --sigma 0.05 --nlayers 0
+
+Atmosphere only simulation:
+  $ python fft_vs_geom_movie.py --diam 8.36 --obscuration 0.61 --sigma 0.0 --nlayers 6 --size 3.0
+
+"""
+
+
+import numpy as np
+import galsim
+
+try:
+    import matplotlib
+    import matplotlib.pyplot as plt
+    import matplotlib.animation as anim
+except ImportError:
+    raise ImportError("This demo requires matplotlib!")
+from distutils.version import LooseVersion
+if LooseVersion(matplotlib.__version__) < LooseVersion('1.2'):
+    raise RuntimeError("This demo requires matplotlib version 1.2 or greater!")
+
+try:
+    from astropy.utils.console import ProgressBar
+except ImportError:
+    raise ImportError("This demo requires astropy!")
+
+def simple_moments(img):
```

Comment:
Added `galsim.utilities.unweighted_moments` and `galsim.utilities.unweighted_shape`, and went through the old tests to make sure these are being used instead of the old `getmoments` helper function (which I removed).

While writing these, I noticed that it wasn't possible to add (subtract) a `PositionD` to (from) a `PositionI`.  I couldn't think of a good reason not to allow this, so I went ahead and implemented type promotion for Positions too.